dms/models/directory.py

```diff
@@ -193,6 +205,52 @@ class DmsDirectory(models.Model):
                 """,
     )
 
+    @api.multi
+    def _get_share_url(self, redirect=False, signup_partner=False, pid=None):
+        self.ensure_one()
+        params = {"access_token": self._portal_ensure_token()}
+        url = "/my/dms/directory/%s" % self.id
+        return "{}?{}".format(url, url_encode(params))
```

Comment:
Why not to override _compute_access_url method instead? 
simila to this: https://github.com/OCA/rma/blob/12.0/rma/models/rma.py#L414